### DISCUSSION TITLE
Why not just rely on staleTime for refetching?

### DISCUSSION THREAD

**Discussion by geekox86** [TanStack/query#3902] [Q&A]
Greetings,

Sorry if this question will sound very basic and obvious. I'm just wondering why do we have multiple refetching options rather than just refetch on stale? What are the use cases that cannot be achieved with the stale time alone?

Many thx!

**Comment by TkDodo** [comment]
refetchOnStale would mean an automatic fetch when the data goes stale - this is pretty much what a refetchInterval would do.

Also, maybe for some queries, you wouldn't consider a reconnect to be a good point in time even if your query is stale, so you set that flag to false. Or, you want to refetch all queries when you reconnect, in which case you can set it to `"always"`.
So while I agree that you can get very far with just setting staleTime, that likely doesn't mean that we should remove the flags.
